File: 0283-move-zeros/move-zeros.ts
function moveZeroes(nums: number[]): void {
  let lastNonZeroIdx = 0;

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] != 0) {
      nums[lastNonZeroIdx] = nums[i];
      lastNonZeroIdx++;
    }
  }
  for (let i = lastNonZeroIdx; i < nums.length; i++) {
    nums[i] = 0;
  }
}

File: 0283-move-zeros/move-zeros.go
func moveZeroes(nums []int) {
	lastNonZeroIdx := 0
	for _, num := range nums {
		if num != 0 {
			nums[lastNonZeroIdx] = num
			lastNonZeroIdx++
		}
	}
	for i := lastNonZeroIdx; i < len(nums); i++ {
		nums[i] = 0
	}
}